src/queries/options.ts

```diff
@@ -12,3 +13,15 @@ export type TextMatchOptions = {
   exact?: boolean;
   normalizer?: NormalizerFn;
 };
+
+export type LegacyQueryOptions = {
+  legacy?: boolean;
```

Comment:
Same remark for this option, not a huge fan of the name but again not sure either it really matters since it should be documented in the migration guide

Reply:
maybe legacyMode? and you can add some jsdoc to explain the meaning a bit more

Reply:
I removed this option since we'll make this mandatory from next release